src/main/java/com/ldtteam/blockout/views/View.java

```diff
@@ -169,6 +176,33 @@ public void setWindow(final Window w)
         }
     }
 
+    @Override
+    public void setSize(final int w, final int h)
+    {
+        super.setSize(w, h);
+
+        // Allow elements to have their size expanded when zero
+        View p = parent;
+        if (p == null) return;
+
+        if (width == 0)
+        {
+            while (p.width == 0 && p.parent != null)
+            {
+                p = p.parent;
+            }
+            width = Math.max(0, p.width - x);
```

Comment:
Also, this makes no sense, the more you go higher in the parent lookup the more this is wrong (p.width/height of anything but this.parent has not connectivity to this element, might be fixable if you count all the offsets during the while loop)

Reply:
Now that is interesting, I think the offset of grandparents simply didn't occur to me.This will indeed create a bug, though a few things would have to be specified in certain ways to produce it. Anyways yeah I'll account for offset in a future PR. Good catch.